sdk/packages/selector-sui/src/adapter.ts

```diff
@@ -0,0 +1,495 @@
+            this._connectionType = ConnectionType.Nightly
+            resolve()
+            return
+          }
+          console.log('open modal')
+          console.log(isStandardConnectedForNetwork(SUI_NETWORK))
+          const recentName = getRecentStandardWalletForNetwork(SUI_NETWORK)
+          console.log(recentName)
+          if (
+            this._eagerConnectForStandardWallets &&
+            recentName !== null &&
+            isStandardConnectedForNetwork(SUI_NETWORK)
+          ) {
+            await this.connectToStandardWallet(recentName, resolve)
+            return
+          }
+
+          this._app.on('userConnected', (e) => {
+            if (this._chosenMobileWalletName) {
+              persistRecentStandardWalletForNetwork(this._chosenMobileWalletName, SUI_NETWORK)
+            } else {
+              clearRecentStandardWalletForNetwork(SUI_NETWORK)
+            }
+            this._accounts = e.publicKeys.map((pk) => createSuiWalletAccountFromString(pk))
+            this.connected = true
+            this._connecting = false
+            this.connecting = false
+            this._connectionType = ConnectionType.Nightly
+            this._modal?.closeModal()
+            resolve()
+          })
+          console.log('modal', this._modal)
+          if (!this._modal) {
+            this._connecting = false
+            this.connecting = false
+            reject(new Error('Wallet not ready'))
+          }
+          // _modal is defined here
+          this._modal!._onClose = () => {
+            if (this._connecting) {
+              this._connecting = false
+              this.connecting = false
+              const error = new Error('Connection cancelled')
+              reject(error)
+            }
+          }
+          // Try open
+          const opened = this._modal!.openModal(this._app!.sessionId, (walletName) => {
+            if (
+              isMobileBrowser() &&
+              !this._walletsList.find((w) => w.name === walletName)?.standardWallet
+            ) {
+              this.connectToMobileWallet(walletName)
+            } else {
+              this.connectToStandardWallet(walletName, resolve)
+            }
+          })
+          // If modal is not opened, reject
+          // This might be caused by SSR
+          if (!opened) {
+            this._connecting = false
+            this.connecting = false
+            const error = new Error('Failed to open modal')
+            reject(error)
+          }
+        } catch (error: unknown) {
+          this._connecting = false
+          this.connecting = false
+
+          reject(error)
+        }
+      }
+
+      innerConnect()
+    })
+  }
+  disconnect = async () => {
+    if (!this.connected) {
+      throw new Error('Wallet not connected')
+    }
+    switch (this._connectionType) {
+      case ConnectionType.Nightly: {
+        // Refresh app session
+        this._app = await AppSui.build(this._appInitData)
```

Comment:
set loading to false before refresh and to true after

Reply:
this is awaited method  so there is not "loading" part

Reply:
ok, then leave it as it is